prov/verbs/src/verbs_info.c

```diff
@@ -1191,15 +1221,25 @@ static int fi_ibv_get_matching_info(uint32_t version,
 				continue;
 		}
 
-		if (!(fi = fi_dupinfo(check_info))) {
-			ret = -FI_ENOMEM;
-			goto err;
-		}
+		if ((check_info->ep_attr->type == FI_EP_MSG) && is_passive) {
+			if (got_passive_info)
+				continue;
 
-		ret = fi_ibv_set_default_info(fi);
-		if (ret) {
-			fi_freeinfo(fi);
-			continue;
+			if (!(fi = fi_ibv_get_passive_info(check_info, hints))) {
+				ret = -FI_ENOMEM;
+				goto err;
+			}
+			got_passive_info = 1;
```

Comment:
The looping around got_passive_info is odd.  It's set to 1 here, but checked above.  Once set, we always continue the loop.  The if check above for FI_EP_MSG is even confusing to me.  If is_passive is set, isn't the ep type always MSG?  It seems that there should be some sort of immediate return from the call (or at least break from the loop) if is_passive is set and a passive_info is found.

Reply:
This function builds the matching fi_info for all endpoint types. If the user hasn't requested a specific EP type, we can't break after we build the fi_info for MSG EP.

`is_passive` is set to 1 when the user requests any address for source and that's independent of EP type requested. The verbs/RDM and verbs/DGRAM code don't handle any address as of now. So we have to do this only for MSG endpoints.

So if the user requests any address we then build just one fi_info corresponding to FI_EP_MSG (even if there are multiple verbs devices). The `got_passive_info` variable tracks if we have built this "passive fi_info". Once it's built we have to skip all other FI_EP_MSG info in check_info (which should probably be renamed to prov_info since that's more appropriate).

Reply:
'passive' implies server side listening, which is MSG EP.  If that's not the intent, I would rename the variables.  Particularly, linking passive with meaning any source is not intuitive and for the common use of 'passive' incorrect.

Reply:
That makes sense. I'll rename the variable to something like any_addr.